### PR TITLE
Add LATEST.md

### LINKED ISSUE #737: Create one more list, organized by date added
We have README, ARCHIVE - what if we had one more list called LATEST which would show the list, organized from new to old — simply showing latest 15 entries should be fine. 

~~Maybe it does not have to contain entire list (top 10, 20, 50?)~~

Thoughts? 

cc @vpeschenkov

### PR DESCRIPTION
**Pull request by vpeschenkov**
Close #737

### CODE REVIEW

**Comment by vpeschenkov** [comment]
@dkhamsing don't merge this. When we decide to merge it, I want to test it once more time.

**Comment by dkhamsing** [comment]
Ok will look and leave a comment if approving.

File: .github/deploy.sh [hunk 1]
```diff
@@ -25,4 +25,44 @@ then
   git commit -m "[auto] [ci skip] Generate APPSTORE"
 fi
 
+if [[ $status == *"LATEST.md"* ]]
+then
+  git add APPSTORE.md
+  git commit -m "[auto] [ci skip] Generate LATEST"
+fi
+
+git push --quiet "https://${GH_TOKEN}@github.com/dkhamsing/open-source-ios-apps" master:master > /dev/null 2>&1
```
Comment (dkhamsing) [inline .github/deploy.sh]: hey @vpeschenkov do we need all these changes??

File: .github/osia_convert.rb [hunk 2]
```diff
@@ -1,268 +1,372 @@
...
-    else
-      !(tags.include?(ARCHIVE_TAG))
-    end
+# File constants 
+README = "README.md"
+ARCHIVE = "ARCHIVE.md"
+APPSTORE = "APPSTORE.md"
+LATEST = "LATEST.md"
+
+# Shows that a project is archived.
```
Comment (dkhamsing) [inline .github/osia_convert.rb]: 💯

File: .github/osia_convert.rb [hunk 3]
```diff
@@ -1,268 +1,372 @@
...
+README = "README.md"
+ARCHIVE = "ARCHIVE.md"
+APPSTORE = "APPSTORE.md"
+LATEST = "LATEST.md"
+
+# Shows that a project is archived. 
+ARCHIVE_TAG = "archive"
+
+NUMBER_OF_LATEST_DAYS = 60
+MINIMUM_LATEST_APPS_COUNT = 15
```
Comment (dkhamsing) [inline .github/osia_convert.rb]: 👍

File: .github/deploy.sh (hunk 1 above)
Comment (vpeschenkov) [reply .github/deploy.sh]: @dkhamsing my bad; didn't pay attention when moving this code

**Comment by dkhamsing** [comment]
Thx for the changes in deploy 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by dkhamsing** [comment]
Do you have README ARCHIVE APPSTORE and LATEST I can review as well?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by vpeschenkov** [comment]
@dkhamsing 

- [README.md](https://github.com/vpeschenkov/open-source-ios-apps/blob/feature/latest-list/README.md)
- [LATEST.md](https://github.com/vpeschenkov/open-source-ios-apps/blob/feature/latest-list/LATEST.md)
- [ARCHIVE.md](https://github.com/vpeschenkov/open-source-ios-apps/blob/feature/latest-list/ARCHIVE.md)
- [APPSTORE.md](https://github.com/vpeschenkov/open-source-ios-apps/blob/feature/latest-list/APPSTORE.md)

**Comment by dkhamsing** [comment]
Have to review more - will post here when done 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: .github/osia_convert.rb [hunk 4]
```diff
@@ -1,268 +1,365 @@
...
+    cat = app["category-ids"]
+    cat.class == Array ? cat.include?(id) : (cat == id)
+  end
+  s.sort_by { |k, v| k["title"].downcase }
+end
+
+##
+# Writes an output to a file.
+# Params
+# - file: A file for writning.
```
Comment (dkhamsing) [inline .github/osia_convert.rb]: Typo, should be `writing`

**Comment by dkhamsing** [comment]
Thanks for cleaning up the script, it’s too long and very hard to read - my fault!
Still reviewing.

File: .github/osia_convert.rb [hunk 5]
```diff
@@ -1,268 +1,365 @@
...
 end
 
-def app_store_total(j)
-  apps = j['projects']
-  s = apps.reject { |x| x['itunes'].nil? }
+##
+#
+def app_store_total(source)
+  apps = source["projects"]
+  apps = select_app_store_apps(apps)
```
Comment (dkhamsing) [inline .github/osia_convert.rb]: Something weird about using `apps` For line 49 and 50

Can this be improved by using `apps = select_app_store_apps(source["projects"])` ?

File: .github/osia_convert.rb [hunk 6]
```diff
@@ -1,268 +1,365 @@
...
 
   count = 1
-  s.each do |x|
-    tags = x['tags']
+  apps.each do |app|
```
Comment (dkhamsing) [inline .github/osia_convert.rb]: Can you leave a comment above for what we are trying to do?

File: .github/osia_convert.rb [hunk 7]
```diff
@@ -1,268 +1,365 @@
...
-    screenshots = a['screenshots']
-    license = a['license']
-
-    t = "[#{name}](#{link})"
-
-    if desc.nil?
-      t << ' '
+##
+#
+def assembly_apps(apps, appstoreonly)
```
Comment (dkhamsing) [inline .github/osia_convert.rb]: Not sure how I feel about `assembly` .. is there an alternative ?

File: .github/osia_convert.rb [hunk 8]
```diff
@@ -1,268 +1,365 @@
...
+  output = ""
+  apps.each do |app|
+    title = "[#{app.title}](#{app.source})"
+
+    if app.description.nil?
+      title << " "
     else
-      t << ": #{desc} " if desc.size>0
+      title << ": #{app.description} " if app.description.size > 0
```
Comment (dkhamsing) [inline .github/osia_convert.rb]: Much better, thx!

File: .github/osia_convert.rb [hunk 9]
```diff
@@ -1,268 +1,365 @@
...
+    output << "  <details>\n\t<summary>"
 
     details = ""
 
-    unless tags.nil?
-      details << '<code>swift</code> ' if tags.include? 'swift'
-      tags.each do |t|
-        details << "<code>#{t.downcase}</code> " if t.downcase != 'swift'
+    unless app.tags.nil?
+      details << "<code>swift</code> " if app.tags.include? "swift"
```
Comment (dkhamsing) [inline .github/osia_convert.rb]: Use of `"swift"` is repeated a lot.. can we make it a constant ?

**Comment by dkhamsing** [comment]
Thanks for trying to cleanup the script. Is there a way to support LATEST with less changes. The original version is very bad but this update makes it very long as well. Any thoughts?

**Comment by dkhamsing** [comment]
This is from the README you provided - something missing
0812BA52-C840-49A0-8BC0-D1E026C96E27

I’m gonna close this PR. Could you do another one without all the changes. We should save a “refactor” of the script for a different PR instead of doing two things at once. Thx

**Comment by vpeschenkov** [comment]
Probably, no, I won't. I don't imagine this without refactoring and changing the code to that way. At least, I won't do this 3 times instead of 1. There is not so much code to be hard to review and fix even if rewriting it from scratch.

**Comment by dkhamsing** [comment]
No worries I can work on this - thx for taking the time. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request closed without merging**